index.js

```diff
@@ -67,28 +67,46 @@ var Node = function (path, opts, disposable) {
         })
       }
     },
-    daemon: function (cb) {
+    startDaemon: function (cb) {
       var t = this
       parseConfig(t.path, function (err, conf) {
-        var running = run(IPFS_EXEC, ['daemon'], {env: t.env})
-        t.pid = running.pid
-        running
+        if (err) return cb(err)
+
+        t.subprocess = run(IPFS_EXEC, ['daemon'], {env: t.env})
           .on('error', function (err) {
             if ((err + '').match('daemon is running')) {
               // we're good
               cb(null, ipfs(conf.Addresses.API))
+            } else if ((err + '').match('non-zero exit code')) {
+              // ignore when kill -9'd
             } else {
               cb(err)
             }
           })
           .on('data', function (data) {
-            var match = (data + '').trim().match(/API server listening on/)
+            var match = (data + '').trim().match(/API server listening on (.*)/)
             if (match) {
-              cb(null, ipfs(conf.Addresses.API))
+              t.apiAddr = match[1]
+              cb(null, ipfs(t.apiAddr))
             }
           })
       })
     },
+    stopDaemon: function () {
+      var t = this
+      var sp = t.subprocess
+
+      if (sp) {
+        sp.kill('SIGHUP')
+        setTimeout(function () {
+          sp.kill('SIGKILL')
```

Comment:
does it need to check? i guess not because this will just fail after.

ideally -- if things exit sooner -- we'd exit sooner. maybe the thing to do is a loop that checks whether the process has died every ~300ms and exits early if so?